msgpack/include/adaptor.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef ARK_MSGPACK_ADAPTOR_HPP 
+#define ARK_MSGPACK_ADAPTOR_HPP 
+
+#include <Ark/Module.hpp>
+#include <msgpack.hpp>
+
+namespace msgpack 
+{
+	MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS) 
+	{
+		namespace adaptor 
+		{
+			// convert ark value
+			template<> struct convert<Value> 
+			{
+				msgpack::object const& operator()(msgpack::object const& o, Value& v) const 
+				{
+					if(o.type == type::NIL)
+					{
+						v = Value(-1);
+					}
+					else if(o.type == type::BOOLEAN)
+					{
+						v = Value(o.via.boolean);
```

Comment:
won't work with the new api, we don't have a constructor accepting booleans

Reply:
Ok